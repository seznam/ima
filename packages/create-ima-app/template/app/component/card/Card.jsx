import './card.less';

export default function Card({ children, title, href }) {
  return (
    <div className='card'>
      <a className='card-title' target='_blank' href={href} rel='noreferrer'>
        <h3>{title} &raquo;</h3>
      </a>
      <p
        className='card-content'
        dangerouslySetInnerHTML={{
          __html: children.replaceAll('{link}', href),
        }}
      />
    </div>
  );
}
