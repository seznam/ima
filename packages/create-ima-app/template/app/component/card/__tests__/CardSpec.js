import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  it('can render', () => {
    const props = {
      title: 'Test card',
      href: 'https://www.seznam.cz',
      children: 'Some content of the card.',
    };

    const wrapper = shallow(<Card {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
