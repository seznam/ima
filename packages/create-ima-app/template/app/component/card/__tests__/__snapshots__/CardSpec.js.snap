// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card can render 1`] = `
<div
  className="card"
>
  <a
    className="card-title"
    href="https://www.seznam.cz"
    rel="noreferrer"
    target="_blank"
  >
    <h3>
      Test card
       Â»
    </h3>
  </a>
  <p
    className="card-content"
    dangerouslySetInnerHTML={
      {
        "__html": "Some content of the card.",
      }
    }
  />
</div>
`;
